libraries/chain/include/eosio/chain/hotstuff/hotstuff.hpp

```diff
@@ -169,6 +168,12 @@ namespace eosio::chain {
                     const bls_public_key& pubkey,
                     const bls_signature& sig);
 
+      fc::sha256 proposal_id()  const { std::lock_guard g(*_mtx); return _proposal_id; };
+      state_t    state()        const { std::lock_guard g(*_mtx); return _state; };
+      votes_t    weak_votes()   const { std::lock_guard g(*_mtx); return _weak_votes; };
+      votes_t    strong_votes() const { std::lock_guard g(*_mtx); return _strong_votes; };
+      std::vector<uint8_t> proposal_digest() const { std::lock_guard g(*_mtx); return _proposal_digest; };
```

Comment:
These all need to be in-sync. Pulling them out individually means they can be modified separately . I think instead you should add a method that returns a `valid_quorum_certificate` and remove all of these.

Reply:
Good idea! Thanks.